docs/tutorials/train_test_overlap.md

```diff
@@ -0,0 +1,312 @@
+# Train-Test Overlap Detection
+
+This tutorial will teach you how to use Marin's train-test overlap detection system to identify contamination between your training datasets and evaluation benchmarks. This is crucial for ensuring the validity of your model evaluations.
+
+## What is Train-Test Overlap?
+
+**Train-test overlap** (also called data contamination) occurs when your training data contains examples that are too similar to your evaluation data. This can lead to:
+
+- **Inflated evaluation scores**: Your model may have "memorized" answers rather than learned to reason
+- **Invalid comparisons**: Results become incomparable to other models trained on clean data
+- **Poor generalization**: Performance doesn't reflect real-world capability
+
+Marin's system detects overlap by finding shared n-grams (sequences of words) between training and evaluation datasets using efficient Bloom filters.
+
+## How It Works
+
+The system uses a two-step process:
+
+1. **Deduplication Step** (`train_test_total.py`):
+   - Builds Bloom filters from training data n-grams
+   - Scans evaluation datasets for matching n-grams
+   - Outputs contaminated examples with overlap statistics
+
+2. **Aggregation Step** (`aggregate_total.py`):
+   - Combines results across all datasets
+   - Generates contamination matrices and CSV summaries
+   - Produces detailed contamination maps
+
+## Quick Start
+
+### Prerequisites
+
+Make sure you have:
+- Marin installed and configured
+- Access to your training and evaluation datasets
+- Ray cluster set up (if running distributed)
+
+### Running on Existing Datasets
```

Comment:
At some point, write 1-2 sentences saying conceptually what happens: For each training set S (e.g., FineWebEdu) and test dataset T (e.g., MMLU), we compute...